### ISSUE TITLE
No mobs in survival worlds due to view area distance

### ISSUE THREAD

**Issue by BugmanBugman** [MarkusBordihn/adaptive_performance_tweaks#17]
**Describe the bug**
My world's are practically empty upon loading up the world. It's a bit weird since even though i used to get decreased mobs, it wouldn't be all of them gone. I used the debug command and looked at my logs, and i think it may have to do the with the view area distance module. I cant figure how to disable it
**To Reproduce**
1. Start off a world with the view area distance on hardcore
2. spawn in
3. notice how there's hardly any mobs (you can use forge entity list to double check)

**Expected behavior**
A decent level of mobs

**Screenshots**
With the mod (look at the right part of the screen listing amount of mobs
![2021-05-27_09 56 10](https://user-images.githubusercontent.com/40730042/119849672-70cfac80-bed2-11eb-99e9-419ee9cb95ee.png)
Without it (alot more mobs)
![2021-05-27_09 30 25](https://user-images.githubusercontent.com/40730042/119849706-775e2400-bed2-11eb-8875-399150ea0335.png)

**Debugging**
https://gist.github.com/BugmanBugman/533c4a1ba31336ae545b191792bc34ea

**Is the issue happening in combination with other mods? (If yes, please provide a full list)**
https://gist.github.com/BugmanBugman/b2176b8dd2a97c008e4ed5d5ded4c031

**Comment by MarkusBordihn** [comment]
Thanks a lot for the detailed report. By comparing your screenshot and debug file most of the spawns would be over 100-160 blocks away from your current position x2917 y67 z3822 (from the screenshot) like:
```
[27May2021 09:51:20.651] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for EntityMoose['Moose'/432692, l='ServerLevel[New World]', x=-2762.50, y=68.00, z=3779.50] in minecraft:overworld.
[27May2021 09:51:20.651] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for ZombieEntity['Zombie'/432693, l='ServerLevel[New World]', x=-2773.50, y=31.00, z=3749.50] in minecraft:overworld.
[27May2021 09:51:20.651] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/432694, l='ServerLevel[New World]', x=-2718.50, y=34.00, z=3686.50] in minecraft:overworld.
[27May2021 09:51:20.651] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for EntityCockroach['Cockroach'/432695, l='ServerLevel[New World]', x=-2697.50, y=36.00, z=3772.50] in minecraft:overworld.
[27May2021 09:51:20.653] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for CreeperEntity['Creeper'/432696, l='ServerLevel[New World]', x=-2704.50, y=45.00, z=3827.50] in minecraft:overworld.
[27May2021 09:51:20.653] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/432698, l='ServerLevel[New World]', x=-2653.50, y=28.00, z=3814.50] in minecraft:overworld.
[27May2021 09:51:20.703] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for IceologerEntity['Iceologer'/432713, l='ServerLevel[New World]', x=-2727.50, y=65.00, z=3658.50] in minecraft:overworld.
[27May2021 09:51:20.704] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for EntityCockroach['Cockroach'/432722, l='ServerLevel[New World]', x=-2753.50, y=44.00, z=3847.50] in minecraft:overworld.
[27May2021 09:51:20.752] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/432736, l='ServerLevel[New World]', x=-2644.50, y=22.00, z=3760.50] in minecraft:overworld.
```

Others entities are below the surface like:
```
[27May2021 09:51:20.104] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for StrayEntity['Stray'/432355, l='ServerLevel[New World]', x=-2681.50, y=27.00, z=3875.50] in minecraft:overworld.
[27May2021 09:51:20.105] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for EntityCockroach['Cockroach'/432360, l='ServerLevel[New World]', x=-2755.50, y=25.00, z=3694.50] in minecraft:overworld.
[27May2021 09:51:20.106] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for EntityReindeer['Reindeer'/432366, l='ServerLevel[New World]', x=-2810.50, y=8.00, z=3748.50] in minecraft:overworld.
BlizzEntity['Blizz'/432202, l='ServerLevel[New World]', x=-2736.50, y=23.00, z=3679.50] in minecraft:overworld.
```

In both cases it's epxected that they are no mobs spawning because the algorithm allows only mob inside 80% of the view area and only below surface if you are "underground" or if you can't see the sky.

I'm currently working on a new algorithm which increase the range a little bit and consider some additional factors.
But in this case Minecraft has not consider to spawn any mob near you.

I will run some test with the mods included in your list, maybe there is a conflict with one of the core mods like RespawnableAnimals, Lithium (Forge), improvedmobs.

Was not really able to reproduce the issue with a new loaded world, after 30secs I got a lot of visible mods (see minimab).

![2021-05-27_23 41 57](https://user-images.githubusercontent.com/12183293/119900938-7ca88680-bf45-11eb-8015-391aa8894a25.png)
![2021-05-27_23 42 04](https://user-images.githubusercontent.com/12183293/119900943-7f0ae080-bf45-11eb-9bd7-0c0acf3cb68f.png)

**Comment by BugmanBugman** [comment]
> Thanks a lot for the detailed report. By comparing your screenshot and debug file most of the spawns would be over 100-160 blocks away from your current position x2917 y67 z3822 (from the screenshot) like:

Oops, I forgot to show me moving around. Those were from me flying around to check for any mobs in the world

You may be right about the incompatible mods. I will test respawnable mobs, improved mobs, and lithium later today.

**Comment by MarkusBordihn** [comment]
The player position will not be tracked in realtime to avoid additional stress for the server.
Which mean flying around will in most case not spawn the same amount of mobs.
This is expected to avoid populating chunks which are only seen for some seconds.

But I found caching issues with the local minecraft server which still keeps track of mobs from a former session.
I have fixed this in version 1.0.3 which could maybe cause the effect you notice.
